### PR TITLE
Add support for build numbers in CI

### PR DESCRIPTION
**Pull request by tristandl**
Modified bundle_version script to accept a provided build number, added tagging to the itunes connect script

### CODE REVIEW

File: components/deployment.sh [hunk 1]
```diff
@@ -26,6 +26,11 @@ function itunes_connect() {
     --scheme "$scheme" \
     $clean_build
 
+  if [[ $build_number ]]; then
+    msg "Tagging build as $build_number"
+    git tag -a build-$build_number -m "Added by ios-tools"
```
Comment (aexmachina) [inline components/deployment.sh]: This is going to create LOTS of tags, depending on how it's used, but that's not clear. Should the circle.yml example be updated? I suggest this argument should only be used when deploying to iTC.

Also, I prefer string interpolation to be done inside double quotes.

File: scripts/bundle_version.sh [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
-bundle_version=$number_of_commits
+bundle_version=$1
+
+if [[ -z $bundle_version ]]; then
+  # If not provided as a command line argument, use the number of commits on the current branch
+  bundle_version=$("$git" rev-list HEAD --count)
+fi
 
+echo $bundle_version
+exit
```
Comment (aexmachina) [inline scripts/bundle_version.sh]: It looks like the usage for this script hasn't been updated either, and I'm guessing `echo ... exit` isn't what it's supposed to do :)

**Comment by tristandl** [comment]
that’s exactly when it will be used (itc)

File: components/deployment.sh (hunk 1 above)
Comment (tristandl) [inline components/deployment.sh]: yep, this is only used inside deploy to itunes

Comment (tristandl) [inline components/deployment.sh]: (that's why it's in the function called `itunes_connect()` 😃 )

File: scripts/bundle_version.sh (hunk 2 above)
Comment (tristandl) [inline scripts/bundle_version.sh]: there isn't any usage for this script. i updated the usage for the execute.sh script though

**Pull request merged**